### PR TITLE
Make the internal fields used by Castle on dynamic proxies private instead of public

### PR DESCRIPTION
**Pull request by spencercw**
See #260.

### CODE REVIEW

File: src/Castle.Core.Tests/DynamicProxy.Tests/SerializableClassTestCase.cs [hunk 1]
```diff
@@ -250,8 +251,8 @@ public void ProxyGenerationOptionsRespectedOnDeserialization()
 			Assert.AreEqual(proxy.GetType(), proxy.GetType().GetMethod("get_Current").DeclaringType);
 			Assert.AreNotEqual(proxy.GetType(), proxy.GetType().GetMethod("CalculateSumDistanceNow").DeclaringType);
 			Assert.AreEqual(proxy.GetType().BaseType, proxy.GetType().GetMethod("CalculateSumDistanceNow").DeclaringType);
-			var options2 =
-				(ProxyGenerationOptions)proxy.GetType().GetField("proxyGenerationOptions").GetValue(null);
+			var options2 = (ProxyGenerationOptions)proxy.GetType().
+				GetField("proxyGenerationOptions", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null);
```
Comment (fir3pho3nixx) [inline src/Castle.Core.Tests/DynamicProxy.Tests/SerializableClassTestCase.cs]: Why is this static a thing? Mostly instance based or have I missed something.

**Comment by fir3pho3nixx** [comment]
From what you have done, it looks good. Could you also update the CHANGELOG.md accordingly.

File: src/Castle.Core.Tests/DynamicProxy.Tests/SerializableClassTestCase.cs (hunk 1 above)
Comment (spencercw) [reply src/Castle.Core.Tests/DynamicProxy.Tests/SerializableClassTestCase.cs]: `proxyGenerationOptions` is a static field (see BaseProxyGenerator.cs). If you don't pass any binding flags then you get `Instance|Static|Public` (see [Type.GetField()](https://referencesource.microsoft.com/#mscorlib/system/type.cs,31ecddf2e1a221ae)), so I need to set the flags manually to get it to return the private field.

**Comment by fir3pho3nixx** [comment]
Just get an entry into the CHANGELOG.md and then we are good to go.

**Comment by spencercw** [comment]
@fir3pho3nixx Enhancement or bugfix do you think? I'm not sure it qualifies as a bugfix because it's not strictly broken.

**Comment by fir3pho3nixx** [comment]
I would say enhancement.

**Comment by spencercw** [comment]
Ok that's done

**Comment by fir3pho3nixx** [comment]
Welcome to the club fella! :) Wish we had more people reporting problems and fixing them like you. Most people tend to use this issue tracker as a substitute for Stack Overflow. 

@jonorossi - will have to merge this because he soul surviving owner ATM. 

Thanks again!

**Comment by spencercw** [comment]
Heh, thanks. :) I probably ought to contribute to these things more than I do. This just happened to be a nice easy one to get into.

**Comment by jonorossi** [comment]
Looks good, yes you should contribute more 😉 .

**Pull request merged**